pkg/ui/handlers.go

```diff
@@ -62,11 +68,24 @@ func (a *API) uiFiles(w http.ResponseWriter, r *http.Request) {
 	a.fileServer.ServeHTTP(w, r)
 }
 
-func NewAPI(fileSystem fs.FS, logger logging.LoggerInterface) *API {
+func (a *API) getCSP(baseURL, kratosPublicURL string) string {
+	b, _ := url.Parse(baseURL)
+	k, _ := url.Parse(kratosPublicURL)
+	if k != nil && b != nil && k.Host != b.Host {
+		// We need to whitelist the kratos URL, because the browser needs to fetch the webauthn.js script from Kratos
+		return fmt.Sprintf("default-src 'self' data: https://*.ubuntu.com; script-src 'self' %v; style-src 'self' 'unsafe-inline'", kratosPublicURL)
+	}
+	return "default-src 'self' data: https://*.ubuntu.com; script-src 'self'; style-src 'self' 'unsafe-inline'"
```

Comment:
suggestion: feel free to disregard the code suggestion but consider the edit to the comment to comply with company guidelines

```suggestion
        additionalScriptURL := ""
	if k != nil && b != nil && k.Host != b.Host {
		// Allowlist the kratos URL to allow the browser needs to fetch the webauthn.js script
		additionalScriptURL := kratosPublicURL
	}
	return fmt.Sprintf("default-src 'self' data: https://*.ubuntu.com; script-src 'self' %v; style-src 'self' 'unsafe-inline'", additionalScriptURL)
```

Reply:
Oh nice, that's better. Thank you